zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/schema/Query.scala

```diff
@@ -17,66 +17,66 @@ trait Statement {}
   * Augments the source data to contain more fields.
   *
   * @param sourceKeys
-  * @param lookupDataset
+  * @param dataset
   * @param lookupKeys
   * @param selectValues
   * @param as
   */
-class LookupStatement(val sourceKeys: Seq[String],
-                      val lookupDataset: String,
-                      val lookupKeys: Seq[String],
-                      val selectValues: Seq[String],
-                      val as: Seq[String]
-                     ) extends Statement {
+case class LookupStatement(val sourceKeys: Seq[String],
```

Comment:
There is one comment below that doc which explains the way we are using case class:
> I often use case classes (of course if there are no complex inheritance) to get auto-generated equals, hashCode, toString, static apply() for shorter initialization, etc.